tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -23,3 +30,53 @@ class resource_type(ArmResourceManager.resource_type):
         service = 'azure.mgmt.keyvault'
         client = 'KeyVaultManagementClient'
         enum_spec = ('vaults', 'list', None)
+
+
+@KeyVault.filter_registry.register('whitelist')
+class WhiteListFilter(ValueFilter):
+    schema = type_schema('whitelist', rinherit=ValueFilter.schema)
+    graph_client = None
+
+    def __init__(self, data, manager=None):
+        super(WhiteListFilter, self).__init__(data, manager)
+        self.op = 'difference'
+
+    def __call__(self, i):
+        if 'access_policies' not in i:
+            client = self.manager.get_client()
+            vault = client.vaults.get(i['resourceGroup'], i['name'])
+            # Retrieve access policies for the keyvaults
+            access_policies = []
+            for policy in vault.properties.access_policies:
+                access_policies.append({
+                    'tenant_id': policy.tenant_id,
+                    'object_id': policy.object_id,
+                    'application_id': policy.application_id,
+                    'permissions': {
+                        'keys': policy.permissions.keys,
+                        'secrets': policy.permissions.secrets,
+                        'certificates': policy.permissions.certificates
+                    }
+                })
+            # Enhance access policies with display_name, object_type and principal_name
+            i['access_policies'] = self.enhance_policies(access_policies)
+        return super(WhiteListFilter, self).__call__(i)
+
+    def enhance_policies(self, access_policies):
+        if self.graph_client is None:
+            s = Session(resource='https://graph.windows.net')
+            self.graph_client = GraphRbacManagementClient(s.get_credentials(), s.get_tenant_id())
+
+        # Retrieve graph objects for all object_id
+        object_ids = [p['object_id'] for p in access_policies]
+        # GraphHelper.get_principal_dictionary returns empty AADObject if not found with graph
+        # or if graph is not available.
+        principal_dics = GraphHelper.get_principal_dictionary(self.graph_client, object_ids)
+
+        for policy in access_policies:
+            aad_object = principal_dics[policy['object_id']]
+            policy['display_name'] = aad_object.display_name
```

Comment:
Same here with casing! Also for consistency with the warning in the Graph helper you should decide on either objectType or aadType and have it be the same in role_assignments. Potentially, since this is very similar code again you could move this into the helper as well.